tiledb/sm/query/sparse_index_reader_base.cc

```diff
@@ -321,10 +321,61 @@ Status SparseIndexReaderBase::read_and_unfilter_coords(
   return Status::Ok();
 }
 
-/** Template specialisation for uint64_t which does result count. */
+template <class BitmapType>
+Status SparseIndexReaderBase::allocate_tile_bitmap(
+    const unsigned dim_num,
+    const Domain* domain,
+    ResultTileWithBitmap<BitmapType>* rt) {
+  auto cell_num = fragment_metadata_[rt->frag_idx()]->cell_num(rt->tile_idx());
+
+  // Bitmap was already computed for this tile.
+  if (rt->bitmap_num_cells != std::numeric_limits<uint64_t>::max()) {
+    return Status::Ok();
+  }
+
+  bool full_overlap = false;
+
+  // For non overlapping ranges, if we have full overlap on any range on
+  // every dimensions, there is no need to allocate a tile bitmap.
+  const bool is_uint8_t = std::is_same<BitmapType, uint8_t>::value;
+  if (is_uint8_t) {
+    full_overlap = true;
+
+    // Get the MBR for this tile.
+    const auto& mbr = fragment_metadata_[rt->frag_idx()]->mbr(rt->tile_idx());
+
+    // See if we have full overlap one dimension at a time.
+    for (unsigned d = 0; d < dim_num; d++) {
+      // No need to compute bitmaps for default dimensions.
+      if (subarray_.is_default(d))
+        continue;
+
+      bool dim_full_overlap = false;
+      const auto dim = domain->dimension(d);
+      const auto& ranges_for_dim = subarray_.ranges_for_dim(d);
+      for (uint64_t r = 0; r < ranges_for_dim.size(); r++) {
```

Comment:
This might be expensive for some of our use cases with millions of ranges.

Reply:
You might be right here, but for now this case is only enabled with single range queries or with a special configuration flag. I plan on doing more performance investigation here in the near future.